### DISCUSSION TITLE
Can subscript be extracted

### DISCUSSION THREAD

**Discussion by fsmosca** [jsvine/pdfplumber#990] [Ask for help with specific PDFs]
I have a [pdf](https://github.com/jsvine/pdfplumber/files/12607752/subscript.pdf) containing a table with subscript text. I tried to extract it, converted it to csv using pandas and save it with utf-8-sig encoding. But it is not working. There is a newline char inserted.

```python
import pdfplumber
import pandas as pd

pdf = pdfplumber.open("subscript.pdf")
table = pdf.pages[0].extract_table()
df = pd.DataFrame(table)
df.to_csv('table.csv', index=False, encoding='utf-8-sig')
print(df)
```

output
```
          0
0       aaa
1     NH\n4
2  C\n10-13
```

How to eliminate the newline char `\n` in table creation?

This should have been NH4 and C10-13.

Note if the subscript in the pdf is really a unicode symbol, then the library can handle it.

**Comment by samkit-jain** [comment]
Hi @fsmosca Appreciate your interest in the library. You'll have to write your own logic to identify what characters are subscript or superscript based on their size attribute

Closing as duplicate of https://github.com/jsvine/pdfplumber/discussions/730

**Reply by fsmosca** [reply]
```
I recommend examining the size and bottom/top attributes of the objects in Page.chars.
```
Is there a sample guide to do that?

**Reply by samkit-jain** [reply]
I would expect the subscript chars to be smallest in size. You can use the following code
```python
import pdfplumber

pdf = pdfplumber.open("file.pdf")
p = pdf.pages[0]

# Smallest character size.
min_size = min(c["size"] for c in p.chars)

# Draw those characters.
im = p.to_image(resolution=200)
im.draw_rects([char for char in p.objects['char'] if char['size'] == min_size])
im.save("image.png", format="PNG")
```
![image](https://github.com/jsvine/pdfplumber/assets/15127115/d66b4a39-fe3f-48af-9499-dc069ba8e530)
